lithium/lithium.py

```diff
@@ -162,46 +89,108 @@ def main():
+    grp_opt.add_argument(
+        "--strategy",
+        default="minimize",
+        choices=["minimize", "remove-pair", "remove-substring", "check-only"],
+        help="reduction strategy to use. default: minimize")
+    grp_add = parser.add_argument_group(description="Additional options for the default strategy")
+    grp_add.add_argument(
+        "--min", type=int,
+        default=1,
+        help="must be a power of two. default: 1")
+    grp_add.add_argument(
+        "--max", type=int,
+        default=pow(2, 30),
+        help="must be a power of two. default: about half of the file")
+    grp_add.add_argument(
+        "--repeat",
+        default="last",
+        choices=["always", "last", "never"],
+        help="Whether to repeat a chunk size if chunks are removed. default: last")
+    grp_add.add_argument(
+        "--chunksize", type=int,
+        default=None,
+        help="Shortcut for repeat=never, min=n, max=n. chunk size must be a power of two.")
+    grp_add.add_argument(
+        "--chunkstart", type=int,
+        default=0,
+        help="For the first round only, start n chars/lines into the file. Best for max to divide n. [Mostly intended for internal use]")
+    grp_add.add_argument(
+        "--repeatfirstround", default=False, action="store_true",
+        help="Treat the first round as if it removed chunks; possibly repeat it.  [Mostly intended for internal use]")
+    grp_add.add_argument(
+        "--maxruntime", type=int,
+        default=None,
+        help="If reduction takes more than n seconds, stop (and print instructions for continuing).")
+    grp_ext = parser.add_argument_group(description="Condition, condition options and file-to-reduce")
+    grp_ext.add_argument(
+        "extra_args",
+        action="append",
+        nargs=argparse.REMAINDER,
+        help="condition [condition options] file-to-reduce")
+
+    args = parser.parse_args()
+
+    tempDir = args.tempdir
+    atom = args.char
+    strategy = args.strategy
+    if args.chunksize:
+        minimizeMin = args.chunksize
+        minimizeMax = args.chunksize
+        repeat = "last"
+    else:
+        minimizeMin = args.min
+        minimizeMax = args.max
+        repeat = args.repeat
+    minimizeChunkStart = args.chunkstart
+    minimizeRepeatFirstRound = args.repeatfirstround
+    if args.maxruntime:
+        stopAfterTime = time.time() + args.maxruntime
+    extra_args = args.extra_args[0]
+
+    if args.testcase is None:
+        try:
+            testcaseFilename = extra_args[-1] # can be overridden by --testcase in processOptions
+        except IndexError:
+            parser.print_help()
+            return False
+    else:
+        testcaseFilename = args.testcase
+    testcaseExtension = os.path.splitext(testcaseFilename)[1]
+
+    if not isPowerOfTwo(minimizeMin) or not isPowerOfTwo(minimizeMax) or len(extra_args) < 1:
```

Comment:
Please restore the old error messages for the 'not isPowerOfTwo' cases. The 'len(extra_args) < 1' case should already be handled above.